File: app/src/main/java/eu/vmpay/random/city/tools/ColorUtils.java
package eu.vmpay.random.city.tools;

public class ColorUtils {
    public static int getFixedColor(String receiver) {
        switch (receiver) {
            case "Yellow":
                return FixedColors.YELLOW.getColor();
            case "Green":
                return FixedColors.GREEN.getColor();
            case "Blue":
                return FixedColors.BLUE.getColor();
            case "Black":
                return FixedColors.BLACK.getColor();
            case "Red":
                return FixedColors.RED.getColor();
            case "White":
                return FixedColors.WHITE.getColor();
            default:
                return FixedColors.UNKNOWN.getColor();
        }
    }
}

File: app/src/main/java/eu/vmpay/random/city/tools/ColorUtils.kt
package eu.vmpay.random.city.tools

import android.graphics.Color
import androidx.annotation.ColorInt

enum class FixedColors(val color: Int) {
    UNKNOWN(Color.GRAY),
    YELLOW(Color.YELLOW),
    GREEN(Color.GREEN),
    BLUE(Color.BLUE),
    BLACK(Color.BLACK),
    WHITE(Color.WHITE),
}

@ColorInt
fun String.getFixedColor(): Int {
    return FixedColors.values().firstOrNull { it.name.equals(this, true) }?.color
            ?: FixedColors.UNKNOWN.color
}
